docs/Routing.md

```diff
@@ -56,12 +61,39 @@ Only render the current route when the URL matches exactly
 
 This is a function that yeilds a [`Query` object](./Queries.md).
 
-### Index Route
+### `indexRoute` (synchronous)
 
 The `indexRoute` parameter is a pared-down `PlatformRoute` definition that
 contains just the `component` that should be rendered when the location/url
 matches the root `path` _exactly_.
 
+### `getIndexRoute` (asynchronous)
+
+A function that, when invoked, will return a Promise that resolves with a
+`Route` object. This can be used for code splitting by using `import()` to
+dynamically import a route to be used as the `exact` index root.
+
+### `routes`
```

Comment:
Or, maybe not? If not, maybe move this below so all the sync/async bits are together.